File: backend/app_tuyen_sinh_django/utils/enums/user_type.py
from enum import Enum


class UserType(Enum):
    STUDENT = "Học sinh"
    PARENT = "Phụ huynh"
    PUPIL = "Sinh viên"
    TEACHER = "Giáo viên"
    FREELANCER = "Thí sinh tự do"
    OTHER = "Khác"

    def __init__(self, name: str) -> None:
        self._value_ = len(__class__)
        self.user_type = name
        __class__.range_val = range(0, self._value_)
        return name

    @staticmethod
    def from_id(input_id: int):
        return next(x for x in __class__ if x.value == input_id)

File: app_tuyen_sinh_flutter/lib/utils/enums/user_type.dart
import 'package:app_tuyen_sinh_flutter/utils/extensions/enum_extension.dart';
import 'package:app_tuyen_sinh_flutter/utils/mixins/enum_id_mixin.dart';

enum UserType with EnumIdMixin {
  student(0, "Học sinh"),
  parent(1, "Phụ huynh"),
  pupil(2, "Sinh viên"),
  teacher(3, "Giáo viên"),
  freelancer(4, "Thí sinh tự do"),
  other(5, "Khác");

  const UserType(this.id, this.name);

  @override
  final int id;
  @override
  final String name;

  static UserType fromId(id) {
    return UserType.values.fromId(id) as UserType;
  }
}
